Eastern-Research-Group/expert-query
# Feature/234 paginate json

## Related Issues:
* [EQ-233](https://jira.epa.gov/browse/EQ-233)
* [EQ-234](https://jira.epa.gov/browse/EQ-234)
* [EQ-236](https://jira.epa.gov/browse/EQ-236)

## Main Changes:
* Removed the JSON format option from the UI.
* Added pagination to API-only JSON responses.
  * This pagination uses a comparison operator on the `objectid` key of each table, which is more consistent than an OFFSET. To simplify this, I added code to ensure the `objectid` is always returned. The user might not need it, since a `nextId` value is returned with every applicable response, but I feel it's okay to always return the unique ID.
  * Refactored some functions in the `data` route implementation.
  * Moved `objectid` to the front of the result set.

## Steps To Test:
1. In _app/server/.env.local_, set `JSON_PAGE_SIZE` to 20 and `MAX_QUERY_SIZE` to 1000. Set the database variables to point to the cloud database.
2. Navigate to http://localhost:9090/attains/data/catchmentCorrespondence?state=AK and confirm the query size exceeded message is returned.
3. Navigate to http://localhost:9090/attains/data/catchmentCorrespondence?organizationId=21AS and confirm a `nextId` field is present in the response.
4. Navigate to http://localhost:9090/attains/data/catchmentCorrespondence?organizationId=21AS&startId=49888219 and confirm there is no `nextId` field in the response.
5. Navigate to http://localhost:9090/attains/data/catchmentCorrespondence?organizationId=21AS&startId=49895501 and confirm that `{ "data": [] }` is returned.

## TODO:
- [x] Review/merge #70 first, since this is branched from it.
